packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -0,0 +1,359 @@
+    const isDateTimeValid = DateTimeUtil.isValidDateTime(this.dateValue, this.timeValue, dateFormat);
+    let updatedDateTime;
+
+    if (isDateTimeValid) {
+      updatedDateTime = DateTimeUtil.updateTime(this.state.dateTime, this.timeValue);
+    }
+
+    this.checkAmbiguousTime(updatedDateTime);
+  }
+
+  checkAmbiguousTime(dateTime) {
+    // To prevent multiple time clarification dialogs from rendered, ensure that it is not open before checking for the ambiguous hour.
+    // ONe situation is when using the right arrow key to move focus from the hour input to the minute input will invoke onBlur and check for ambiguous hour.
+    // If the hour is ambiguous, the dialog would display and steals focus from the minute input, which again will invoke onBlur and check for ambiguous hour.
+    if (this.state.isTimeClarificationOpen) {
+      return;
+    }
+
+    let isDateTimeAmbiguous = false;
+    const isOldTimeAmbiguous = this.state.isAmbiguousTime;
+    if (dateTime && dateTime.isValid()) {
+      const tempDateTime = dateTime.clone();
+      isDateTimeAmbiguous = DateTimeUtil.checkAmbiguousTime(tempDateTime);
+    }
+
+    this.setState({
+      isAmbiguousTime: isDateTimeAmbiguous,
+      isTimeClarificationOpen: isDateTimeAmbiguous && !isOldTimeAmbiguous,
+    });
+  }
+
+  handleDateChange(event, date) {
+    if (event.type === 'change') {
+      this.dateValue = event.target.value;
+    }
+
+    let updatedDateTime;
+
+    if (DateTimeUtil.isValidDate(date, 'YYYY-MM-DD')) {
+      const previousDateTime = this.state.dateTime ? this.state.dateTime.clone() : null;
+      updatedDateTime = DateTimeUtil.syncDateTime(previousDateTime, date, this.timeValue);
+
+      if (DateTimeUtil.isValidTime(this.timeValue)) {
+        this.timeValue = DateTimeUtil.formatISODateTime(updatedDateTime.format(), 'HH:mm');
+      }
+    }
+
+    this.handleChange(event, updatedDateTime);
+  }
+
+  handleDateChangeRaw(event, date) {
+    this.dateValue = event.target.value;
+
+    this.handleChangeRaw(event, date);
+  }
+
+  handleTimeChange(event, time) {
+    this.timeValue = time;
+    const dateFormat = DateUtil.getFormatByLocale(this.context.intl.locale);
+    const validDate = DateTimeUtil.isValidDate(this.dateValue, dateFormat);
+    const validTime = DateTimeUtil.isValidTime(this.timeValue);
+    const previousDateTime = this.state.dateTime ? this.state.dateTime.clone() : null;
+
+    // If both date and time are valid, check if the time is the missing hour and invoke onChange.
+    // If the date is valid but time is invalid, the time in the dateTime state needs to be cleared and render.
+    if (validDate && validTime) {
+      const updatedDateTime = DateTimeUtil.updateTime(previousDateTime, time);
+
+      if (event.keyCode === keyCodes.ARROWDOWN &&
+        previousDateTime && updatedDateTime && previousDateTime.format() === updatedDateTime.format()) {
+        updatedDateTime.add(-1, 'hours');
```

Comment:
Is there a reason you're adding -1 hours instead of subtracting 1?

Reply:
I agree that subtracting 1 hour is easier to understand than adding -1 hour. https://github.com/cerner/terra-core/commit/7f704a5dd8211fc1627f14d09ab2fa2b9f92f5ec